### PR TITLE
Enhance Numeric Data Inspection and Introduce Positive/Negative Filtering

### PR DESCRIPTION
**Pull request by MooooCat**
Enhance NumericInspector and Implement PositiveNegativeFilter

## Description
This PR introduces significant enhancements to the Synthetic Data Generator (SDG) framework, specifically in the `NumericInspector` class and the addition of a new `PositiveNegativeFilter` class. The `NumericInspector` has been updated to support the identification of both positive and negative numeric columns, improving the quality of synthetic data generation. The `PositiveNegativeFilter` class is designed to filter data based on the positivity or negativity of values in specified columns, ensuring that the integrity of the data is maintained during processing.

Key changes include:
- Updated `NumericInspector` to classify columns as positive or negative based on defined thresholds.
- Introduced `PositiveNegativeFilter` to enforce positivity or negativity constraints on specified columns during data processing.
- Added comprehensive test cases to validate the functionality of the new filter and the updated inspector.

## Motivation and Context
The motivation behind these changes is to enhance the data quality assurance mechanisms within the SDG framework. By allowing the identification of positive and negative columns, we can ensure that the synthetic data generated meets specific criteria, which is crucial for various applications such as model training and data sharing. This change addresses the need for more robust data validation and filtering capabilities, ultimately leading to better performance and reliability of the generated synthetic data.

## How has this been tested?
The changes have been thoroughly tested using a dedicated test suite. The following tests were performed:
- Unit tests for the updated `NumericInspector` to ensure correct identification of positive and negative columns.
- Integration tests for the `PositiveNegativeFilter` to verify that it correctly filters data based on the positivity and negativity of values in specified columns.
- Tests included checks for the integrity of mixed columns, ensuring they remain unchanged during filtering processes.
- All tests were executed in a controlled environment using pytest, and all assertions passed successfully.

## Types of changes

- [ ] Maintenance (no change in code, maintain the project's CI, docs, etc.)
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by MooooCat** [comment]
@jalr4ever  Please help me review this PR.

File: sdgx/data_models/inspectors/numeric.py [hunk 1]
```diff
@@ -14,69 +14,132 @@ class NumericInspector(Inspector):
...
     """
 
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
-        self.int_columns: set[str] = set()
-        self.float_columns: set[str] = set()
         self._int_rate = 0.9
         self.df_length = 0
 
     def _is_int_column(self, col_series: pd.Series):
```
Comment (jalr4ever) [inline sdgx/data_models/inspectors/numeric.py:58]: More return type hint would be better:

`def _is_int_column(self, col_series: pd.Series) -> bool:`

File: sdgx/data_models/inspectors/numeric.py [hunk 2]
```diff
@@ -14,69 +14,132 @@ class NumericInspector(Inspector):
...
+            col_series (pd.Series): The column series to be inspected.
+
+        Returns:
+            bool: True if the column is predominantly negative, False otherwise.
+        """
+        return self._is_positive_or_negative_column(
+            col_series, self.negative_threshold, lambda x: x < 0
+        )
 
     def fit(self, raw_data: pd.DataFrame, *args, **kwargs):
```
Comment (jalr4ever) [inline sdgx/data_models/inspectors/numeric.py:144]: Both two `for` loop can merge to one, small refer shoot:

```python
def fit(self, raw_data: pd.DataFrame, *args, **kwargs):
    self.int_columns = set()
    self.float_columns = set()
    self.positive_columns = set()
    self.negative_columns = set()
    self.df_length = len(raw_data)
    
    for col in raw_data.columns:
        if raw_data[col].dtype in ['int64', 'float64']:
            if self._is_int_column(raw_data[col]):
                self.int_columns.add(col)
            else:
                self.float_columns.add(col)
            
            if self._is_positive_column(raw_data[col]):
                self.positive_columns.add(col)
            elif self._is_negative_column(raw_data[col]):
                self.negative_columns.add(col)
    
    self.ready = True

```

**Review by jalr4ever — COMMENTED** [review]
The suggestions in the PR are optional changes.

File: sdgx/data_models/inspectors/numeric.py (hunk 2 above)
Comment (MooooCat) [reply sdgx/data_models/inspectors/numeric.py:144]: I have updated the reviewed content in the code for this PR.

**Comment by MooooCat** [comment]
Modified the code according to the suggestions in the code review, and all unit tests have passed.

**Review by jalr4ever — APPROVED** [review]

File: sdgx/data_models/inspectors/numeric.py [hunk 3]
```diff
@@ -14,69 +14,132 @@ class NumericInspector(Inspector):
 
     This class is a subclass of `Inspector` and is designed to provide methods for inspecting
     and analyzing numeric data. It includes methods for detecting int or float data type.
+
+    In August 2024, we introduced a new feature that will continue to judge the positivity or
```
Comment (Wh1isper) [inline sdgx/data_models/inspectors/numeric.py:18]: Perhaps we should indicate the PR and release version here, rather than the date?

Comment (MooooCat) [reply sdgx/data_models/inspectors/numeric.py:18]: Good idea, I also have another branch in development,  I'll release after merging another PR. Due to various reasons, we haven't released a new version for a long time :(

Comment (Wh1isper) [reply sdgx/data_models/inspectors/numeric.py:18]: Nerver mind, thanks for your work!

**Pull request merged**